langkit/templates/python_api/module_py.mako

```diff
@@ -1856,3 +1857,67 @@ def _field_address(struct, field_name):
 #
 
 ${exts.include_extension(ctx.ext("python"))}
+
+#
+# App base classes
+#
+
+<% short_name = ctx.short_name.lower if ctx.short_name else 'mdl' %>
+
+class App(object):
+    """
+    Base class to regroup logic for an app. We use a class so that
+    specific languages implementations can add specific arguments and
+    processing by overriding:
+
+    - add_arguments to add arguments to the argparse.Parser instance
+
+    - create_unit_provider to return a custom unit provider to be used by the
+      AnalysisContext.
+    """
+
+    @property
+    def description(self):
+        """
+        Description for this app. Needs to be subclassed.
+        """
+        pass
+
+    def __init__(self):
+        self.parser = argparse.ArgumentParser(
+            description=self.description
+        )
+        self.parser.add_argument('files', nargs='+', help='Files')
+        self.add_arguments()
+        self.args = self.parser.parse_args()
+        self.ctx = AnalysisContext(
+            'utf-8', with_trivia=True,
+            unit_provider=self.create_unit_provider()
+        )
+
+    def add_arguments(self):
+        """
+        Hook for subclasses to add arguments to self.parser. Default
+        implementation does nothing.
+        """
+        pass
+
+    def create_unit_provider(self):
+        """
+        Hook for subclasses to return a custom unit provider.
+        Default implementation returns None.
+        """
+        return None
+
+    def process_files(self):
```

Comment:
Looks like subclasses need to use this method: could you document it?

Reply:
I made the call to `process_files` automatic